### PR TITLE
[CCXDEV-11713] Added capability to send compressed messages

### PR DESCRIPTION
**Pull request by Jakub007d**
# Description

Added capability to send compressed messages to kafka topic

Fixes # (issue)

## Type of change

- New feature (non-breaking change which adds functionality)

## Checklist
* [ ] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: ccx_messaging/publishers/kafka_publisher.py [hunk 1]
```diff
@@ -41,6 +42,7 @@ def __init__(self, outgoing_topic: str, kafka_broker_config: dict = None, **kwar
             raise CCXMessagingError("outgoing_topic should be a str")
 
         self.topic = outgoing_topic
+        self.compression = kwargs.get("compression")
```
Comment (juandspy) [inline ccx_messaging/publishers/kafka_publisher.py:45]: shall we parse this to a bool somehow? imagine the user configures the YAML value as "false". If self.compression treats it as a string, `if self.compression:` will be triggered

Comment (Jakub007d) [reply ccx_messaging/publishers/kafka_publisher.py:45]: because it is able to convert "" to false nil to false "true" to true etc

Comment (joselsegura) [reply ccx_messaging/publishers/kafka_publisher.py:45]: You can put a default to False, something like

`bool(kwargs.get("compression", False))`

**Review by tisnik — APPROVED** [review]
ok, just pls fix the issue @juandspy pointed to

**Pull request closed without merging**